packages/design-system/src/styles/components/_Drawer.scss

```diff
@@ -148,5 +145,6 @@
 }
 
 .ds-c-drawer__footer-title {
+  font-weight: var(--font-weight-bold);
```

Comment:
Needed to specify as the original footer title was relying on the default user agent browser styles before.